assignment02/worldgen.c

```diff
@@ -0,0 +1,174 @@
+#include <stdio.h>
+#include <time.h>
+#include <stdbool.h>
+#include <assert.h>
+#include "worldgen.h"
+#include "voronoi.h"
+
+void print_map(terrain_map_t *map) // TODO: take grid not map
+{
+    for(int i = 0; i < GRID_HEIGHT; i++) {
+        for(int j = 0; j < GRID_WIDTH; j++) {
+            char *color = "";
+            switch (map->grid[i][j]) {
+                case '%':
+                    color = MAGENTA;
+                    break;
+                case '.':
+                    color = YELLOW;
+                    break;
+                case '~':
+                    color = BLUE;
+                    break;
+                case '^':
+                    color = GREEN;
+                    break;
+                case ':':
+                    color = GREEN;
+                    break;
+                case '#':
+                    color = BROWN;
+                    break;
+                case 'C':
+                    color = RED;
+                    break;
+                case 'M':
+                    color = LBLUE;
+                    break;
+                default:
+                    // unreachable
+                    fprintf(stderr, "unhandled color case\n");
+                    break;
+            }
+            printf("%s%c", color, map->grid[i][j]);
+        }
+        putchar('\n');
+    }
+}
+
+static inline void get_input(int n, char* buf)
+{
+    fgets(buf, n, stdin);
+}
+
+Location_t handle_input(int n, char* buf, Location_t loc, Gates_t* gates, bool* quit)
```

Comment:
it really needs to be figured out how location is internally represented, I think it is really confusing looking at this (especially for fly which ik is a special case). It could just be some comments